src/Microsoft.TestPlatform.CommunicationUtilities/Messages/MessageType.cs

```diff
@@ -13,31 +20,39 @@ public static class MessageType
     /// <summary>
     /// The session start.
     /// </summary>
+    [ProtocolVersion(Version0, typeof(void), Description = "Not used.")]
     public const string SessionStart = "TestSession.Start";
 
     /// <summary>
     /// The session end.
     /// </summary>
+    [ProtocolVersion(Version0, typeof(void))]
     public const string SessionEnd = "TestSession.Terminate";
 
     /// <summary>
     /// The is aborted.
     /// </summary>
+    [ProtocolVersion(Version0, typeof(void), Description = "Is sent to testhost, and then ignored.")]
     public const string SessionAbort = "TestSession.Abort";
 
     /// <summary>
     /// The session connected.
     /// </summary>
+    [ProtocolVersion(Version0, typeof(void), Description = "Is sent from DesignMode client (vstest.console) back to IDE, to signal that we are ready to recieve requests.")]
     public const string SessionConnected = "TestSession.Connected";
 
     /// <summary>
     /// Test Message
     /// </summary>
+    [ProtocolVersion(Version0, typeof(TestMessagePayload), Description = "A log message or similar message sent back from console or testhost.")]
     public const string TestMessage = "TestSession.Message";
 
     /// <summary>
     /// Protocol Version
     /// </summary>
+    [ProtocolVersion(Version1, typeof(int), Description = @"Used for version handshake from IDE to vstest.console and from vstest.console to testhost,
```

Comment:
Message archivists are welcome to help me annotate this. Probably putting the notes in the summary comment might be better. 

I am thinking about creating some more static types here that will hold the message + type, so I can then use that to infer <T> automatically in all the usages of the message.